mesheryctl/internal/cli/root/system/channel_test.go

```diff
@@ -0,0 +1,121 @@
+package system
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+var b *bytes.Buffer
+
+func SetupContextEnv(t *testing.T) {
+	path, err := os.Getwd()
+	if err != nil {
+		t.Error("unable to locate meshery directory")
+	}
+	viper.Reset()
+	viper.SetConfigFile(path + "/../../../../pkg/utils/TestConfig.yaml")
+	//fmt.Println(viper.ConfigFileUsed())
+	err = viper.ReadInConfig()
+	if err != nil {
+		t.Errorf("unable to read configuration from %v, %v", viper.ConfigFileUsed(), err.Error())
+	}
+
+	mctlCfg, err = config.GetMesheryCtl(viper.GetViper())
+	if err != nil {
+		t.Error("error processing config", err)
+	}
+}
+
+func SetupFunc(t *testing.T) {
+	//fmt.Println(viper.AllKeys())
+	b = bytes.NewBufferString("")
+	logrus.SetOutput(b)
+	logrus.SetFormatter(&utils.OnlyStringFormatterForLogrus{})
+	SystemCmd.SetOut(b)
+}
+
+func BreakupFunc(t *testing.T) {
+	viewCmd.Flags().VisitAll(setFlagValueAsUndefined)
+}
+
+func setFlagValueAsUndefined(flag *pflag.Flag) {
```

Comment:
Ideally, this is defined as a constant, not a function.

Reply:
Visit all accepts a function with the specified format. That is why it's a function.

Reply:
Clarification: the variable defined in ‘ setFlagValueAsUndefined’ doesn’t need to be set in a function, but can just be a declared constant